core/src/main/scala/weaponregex/internal/parser/ParserJS.scala

```diff
@@ -73,6 +73,15 @@ private[weaponregex] class ParserJS private[parser] (pattern: String, val flags:
     else
       P(preDefinedCharClass | metaCharacter | range | quoteChar | charClassCharLiteral)
 
+  /** Parse a group name
+    * @return
+    *   the parsed name string
+    * @example
+    *   `"name1"`
+    */
+  override def groupName[A: P]: P[String] =
+    P(CharIn("a-z", "A-Z", "_") ~ CharIn("a-z", "A-Z", "0-9", "_").rep).!
```

Comment:
Can a group also start with an underscore?

Reply:
Yes it can

Reply:
[regex101](https://regex101.com/r/OAekVc) says it can.